doc/platform_api/CMIS_and_C-CMIS_support_for_ZR.md

```diff
@@ -0,0 +1,650 @@
+            elif vdm_info_dict[1] == 'F16':
+                vdm_value_int = struct.unpack('>H',vdm_value_raw)[0]
+                vdm_value = get_F16(vdm_value_int)
+                vdm_thrsh_high_alarm_int = struct.unpack('>H', vdm_thrsh_high_alarm_raw)[0]
+                vdm_thrsh_low_alarm_int = struct.unpack('>H', vdm_thrsh_low_alarm_raw)[0]
+                vdm_thrsh_high_warn_int = struct.unpack('>H', vdm_thrsh_high_warn_raw)[0]
+                vdm_thrsh_low_warn_int = struct.unpack('>H', vdm_thrsh_low_warn_raw)[0]
+                vdm_thrsh_high_alarm = get_F16(vdm_thrsh_high_alarm_int)
+                vdm_thrsh_low_alarm = get_F16(vdm_thrsh_low_alarm_int)
+                vdm_thrsh_high_warn = get_F16(vdm_thrsh_high_warn_int)
+                vdm_thrsh_low_warn = get_F16(vdm_thrsh_low_warn_int)
+            else:
+                continue
+
+        if vdm_info_dict[0] not in VDM_Page_data:
+            VDM_Page_data[vdm_info_dict[0]] = {VDM_lane[index]+1: [vdm_value,
+                                                                   vdm_thrsh_high_alarm,
+                                                                   vdm_thrsh_low_alarm,
+                                                                   vdm_thrsh_high_warn,
+                                                                   vdm_thrsh_low_warn]}
+        else:
+            VDM_Page_data[vdm_info_dict[0]][VDM_lane[index]+1] = [vdm_value,
+                                                                  vdm_thrsh_high_alarm,
+                                                                  vdm_thrsh_low_alarm,
+                                                                  vdm_thrsh_high_warn,
+                                                                  vdm_thrsh_low_warn]
+    return VDM_Page_data
+
+def get_VDM(port):
+    vdm_page_supported_raw = read_reg_from_dict(port, Page2Fh.VDM_SUPPORT) & 0x3
+    vdm_page_supported = Data_Type_Dict.VDM_SUPPORTED_PAGE[vdm_page_supported_raw]
+    VDM = {}
+    # Bit 7, freeze all PMs for reporting
+    write_reg_from_dict(port, Page2Fh.FREEZE_REQUEST, 128)
+    time.sleep(1)
+    for page in vdm_page_supported:
+        VDM_current_page = get_VDM_page(port, page)
+        VDM.update(VDM_current_page)
+    write_reg_from_dict(port, Page2Fh.FREEZE_REQUEST, 0)
+    return VDM
+```
+
+#### Get C-CMIS PM
+- get_PM
+
+Sample code to read C-CMIS defined PMs:
+
+```
+def get_PM(port):
+
+    write_reg_from_dict(port, Page2Fh.FREEZE_REQUEST, 128)
+    time.sleep(1)
+
+    PM_dict = {}
+    rx_bits_pm = read_reg_from_dict(port, Page34h.RX_BITS_PM)
+    rx_bits_subint_pm = read_reg_from_dict(port, Page34h.RX_BITS_SUB_INTERVAL_PM)
+    rx_corr_bits_pm = read_reg_from_dict(port, Page34h.RX_CORR_BITS_PM)
+    rx_min_corr_bits_subint_pm = read_reg_from_dict(port, Page34h.RX_MIN_CORR_BITS_SUB_INTERVAL_PM)
+    rx_max_corr_bits_subint_pm = read_reg_from_dict(port, Page34h.RX_MAX_CORR_BITS_SUB_INTERVAL_PM)
```

Comment:
These register values are stored in BIG-endian format on the module's EEPROM. We should convert to LITTLE-endian.

Reply:
Hi Prince, please refer to https://www.oiforum.com/wp-content/uploads/OIF-C-CMIS-01.1.pdf Table 13. The MSB is at lower address, shouldn't it be big-endian?

Reply:
may be read_reg() is doing the conversion the already.